### ISSUE TITLE
Gitlab - abaplint for commited objects only

### ISSUE THREAD

**Issue by Zeatynis** [abaplint/abaplint#2456]
Hi, I don't know if this is the right approach but I managed to do a pipeline in GitLab which is triggered for every commit and it gives list of issues only in these, commited, objects.

I know this probably looks messy for DevOps but I am not one.

```
abaplint_commit_only:
    image: abaplint/abaplint
    stage: test
    script:
        - git diff-tree --no-commit-id --name-only -r $CI_MERGE_REQUEST_TARGET_BRANCH_SHA -r $CI_COMMIT_SHA > commited.txt
        - git ls-files > files.txt
        - grep -Fvxf commited.txt files.txt > exclude.txt
        - sed -e 's/^/"/' -e 's/$/"/' -e '$!s/$/,/' exclude.txt > exclude_final.txt
        - wget https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
        - chmod +x jq-linux64
        - sed $'/###NOISSUES###/{r exclude_final.txt\nd}' abaplint_template | ./jq-linux64 . > abaplint_commited.json
        # - cat abaplint_commited.json
        - abaplint abaplint_commited.json --outformat junit --outfile report_commit.xml
    artifacts:
        reports:
            junit: report_commit.xml

```

abaplint_template fragment which is relevant
```
...
"global": {
        "exclude": [],
        "noIssues" : [
        ###NOISSUES###
        ],
...
```

**Comment by larshp** [comment]
closing
